src/pyfmi/fmi_util.pyx

```diff
@@ -356,10 +376,17 @@ cpdef convert_sorted_vars_name_desc(list sorted_vars):
     cdef char *ctmp_name
     cdef char *ctmp_desc
 
-    for i in range(items):
-        var = sorted_vars[i]
-        tmp_name = encode(var.name)
-        tmp_desc = encode(var.description)
+    for i in range(items+nof_diag_params+nof_diag_vars):
+        if i < items:
+            var = sorted_vars[i]
+            tmp_name = encode(var.name)
+            tmp_desc = encode(var.description)
+        elif i < items+nof_diag_params:
+            tmp_name = encode(diag_param_names[i-items][0])
+            tmp_desc = encode(diag_param_names[i-items][1])
+        else:
+            tmp_name = encode(diag_vars[i-items-nof_diag_params][0])
+            tmp_desc = encode(diag_vars[i-items-nof_diag_params][1])
```

Comment:
This looks pretty fragile. Perhaps it would be better to loop over a generator?

```suggestion
    for tmp_name, tmp_desc in itertools.chain(map(lambda var: (var.name, var.description), sorted_vars), 
                                              diag_param_names, 
                                              diag_vars):
```
It will not create any copies of the data structures involved. Requires importing itertools, though.

Reply:
Fixed